src/data/bmg_plate.py

```diff
@@ -0,0 +1,109 @@
+import os
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+class Plate:
+    """
+    Class representing a plate with values resulting from an HTS experiment
+    """
+    def __init__(self, filepath: str) -> None:
+        """
+        :param filepath: path to the file consisting plate
+        """
+        self.filepath = filepath
+        self.barcode = filepath.split('/')[-1].split('.')[0].split('\\')[-1]
+        self.plate_array = self.parse_file()
+        self.control_statistics()
+        self.z_factor = 1 - (3*(self.std_pos+self.std_neg) /
+                             (self.mean_neg-self.mean_pos))
+
+    def parse_file(self) -> np.array:
+        """
+        Read data from txt file to np.array
+
+        :return: array with plate values
+        """
+        plate = np.zeros(shape=(16, 24))
+        with open(self.filepath) as f:
+            for i, line in enumerate(f.readlines()):
+                # handle different formatting (as in FIREFLY)
+                if i == 0 and 'A' not in line:
+                    df = pd.read_csv(self.filepath, header=None)
+                    plate = df.to_numpy()
+                    break
+                well, value = line.split()
+                i, j = self.well_to_ids(well)
+                plate[i, j] = value
+        return plate
+
+    def well_to_ids(self, well_name: str) -> tuple[int, int]:
+        """
+        Helper method to map well name to index
+
+        :param well_name: well name in format {letter}{number} (e.g. A10)
+        to be transformed
+        """
+        head = well_name.rstrip('0123456789')
```

Comment:
maybe use regex here: `head = re.sub(r'\d*$', '', well_name)`

Reply:
tbh I find the rstrip more readable in this case

Reply:
+ rstrip is faster

Reply:
Ok then 👍👍